src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/ReferenceCode.kt

```diff
@@ -1,38 +1,54 @@
 package uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa
 
 import jakarta.persistence.Column
-import jakarta.persistence.DiscriminatorColumn
+import jakarta.persistence.Convert
+import jakarta.persistence.Embeddable
+import jakarta.persistence.EmbeddedId
 import jakarta.persistence.Entity
-import jakarta.persistence.Id
-import jakarta.persistence.IdClass
 import jakarta.persistence.Inheritance
 import org.hibernate.Hibernate
+import org.hibernate.type.YesNoConverter
 import java.io.Serializable
 import java.util.Objects
 
 @Entity(name = "REFERENCE_CODES")
-@DiscriminatorColumn(name = "domain")
+@org.hibernate.annotations.DiscriminatorFormula(value = "domain")
 @Inheritance
-@IdClass(ReferenceCode.Pk::class)
 abstract class ReferenceCode(
 
-  @Id
   @Column(insertable = false, updatable = false)
   open val domain: String,
 
-  @Id
+  @Column(insertable = false, updatable = false)
   open val code: String,
```

Comment:
maybe eventually we just get rid of the double mapping, we could even add a "getter" to accessed the code and domain from within the `id` field